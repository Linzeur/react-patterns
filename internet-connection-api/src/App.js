import React from "react";
import ListEvents from "./events-list";
import useInternetConnection from "./use-internet-connection";
import InternetConnectionChildren from "./internet-connection-function-children";
import InternetConnectionRender from "./internet-connection-render";
import withInternetConnection from "./internet-connection-hoc";
import InternetConnectionContainerPresentational from "./containers/internet-connection-container-presentational";

function App({ pattern }) {
  const title = <h1>List of execution of Page Internet Connection API </h1>;
  const statusHooks = useInternetConnection();
  switch (pattern) {
    case "custom-hooks":
      return <ListEvents title={title} status={statusHooks} />;
    case "function-children":
      return (
        <InternetConnectionChildren>
          {status => {
            return <ListEvents title={title} status={status} />;
          }}
        </InternetConnectionChildren>
      );
    case "render-props":
      return (
        <InternetConnectionRender
          render={status => {
            return <ListEvents title={title} status={status} />;
          }}
        />
      );
    case "high-order-component":
      function HOC({ statusHOC }) {
        return <ListEvents title={title} status={statusHOC} />;
      }
      const InternetConnectionHOC = withInternetConnection(HOC);
      return <InternetConnectionHOC />;
    case "container-presentational":
      return <InternetConnectionContainerPresentational title={title} />;
    default:
      return <ListEvents title={title} status={statusHooks} />;
  }
}

export default App;
