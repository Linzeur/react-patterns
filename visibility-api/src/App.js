import React from "react";
import ListEvents from "./events-list";
import useVisibility from "./use-visibility";
import VisibilityChildren from "./visibility-function-children";
import VisibilityRender from "./visibility-render";
import withVisibility from "./visibility-hoc";
import VisibilityContainerPresentational from "./containers/visibility-container-presentational";

function App({ pattern }) {
  const title = <h1>List of execution of Page visibility API </h1>;
  const listEventsHooks = useVisibility();
  switch (pattern) {
    case "custom-hooks":
      return <ListEvents title={title} listEvents={listEventsHooks} />;
    case "function-children":
      return (
        <VisibilityChildren>
          {listEvents => {
            return <ListEvents title={title} listEvents={listEvents} />;
          }}
        </VisibilityChildren>
      );
    case "render-props":
      return (
        <VisibilityRender
          render={listEvents => {
            return <ListEvents title={title} listEvents={listEvents} />;
          }}
        />
      );
    case "high-order-component":
      function HOC({ listEventsHOC }) {
        return <ListEvents title={title} listEvents={listEventsHOC} />;
      }
      const VisibilityHOC = withVisibility(HOC);
      return <VisibilityHOC />;
    case "container-presentational":
      return <VisibilityContainerPresentational title={title} />;
    default:
      return <ListEvents title={title} listEvents={listEventsHooks} />;
  }
}

export default App;
