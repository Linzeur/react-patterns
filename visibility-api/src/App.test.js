import React from "react";
import { render } from "@testing-library/react";

import App from "./App";

const listEvents = [
  { date: new Date("06/15/2019"), value: false },
  { date: new Date("06/18/2019"), value: false },
  { date: new Date("06/19/2019"), value: true }
];
test("Test app component using pattern Function as Children", () => {
  const { asFragment } = render(<App pattern={"function-children"} />);

  expect(asFragment()).toMatchSnapshot();
});

test("Test event list component using pattern Render Props", () => {
  const { asFragment } = render(<App pattern={"render-props"} />);

  expect(asFragment()).toMatchSnapshot();
});

test("Test event list component using pattern High Order Components", () => {
  const { asFragment } = render(<App pattern={"high-order-component"} />);

  expect(asFragment()).toMatchSnapshot();
});

test("Test event list component using pattern Container Presentational", () => {
  const { asFragment } = render(<App pattern={"container-presentational"} />);

  expect(asFragment()).toMatchSnapshot();
});

test("Test event list component using pattern Custom Hooks", () => {
  const { asFragment } = render(<App pattern={"custom-hooks"} />);

  expect(asFragment()).toMatchSnapshot();
});
