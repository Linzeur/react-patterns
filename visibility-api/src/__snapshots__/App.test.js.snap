// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test app component using pattern Function as Children 1`] = `
<DocumentFragment>
  <h1>
    List of execution of Page visibility API 
  </h1>
  <ul />
</DocumentFragment>
`;

exports[`Test event list component using pattern Container Presentational 1`] = `
<DocumentFragment>
  <h1>
    List of execution of Page visibility API 
  </h1>
  <ul />
</DocumentFragment>
`;

exports[`Test event list component using pattern Custom Hooks 1`] = `
<DocumentFragment>
  <h1>
    List of execution of Page visibility API 
  </h1>
  <ul />
</DocumentFragment>
`;

exports[`Test event list component using pattern High Order Components 1`] = `
<DocumentFragment>
  <h1>
    List of execution of Page visibility API 
  </h1>
  <ul />
</DocumentFragment>
`;

exports[`Test event list component using pattern Render Props 1`] = `
<DocumentFragment>
  <h1>
    List of execution of Page visibility API 
  </h1>
  <ul />
</DocumentFragment>
`;
